js/add-site.js

```diff
@@ -1,38 +1,70 @@
 
+/**
+ * Initiate search functionality
+ */
+
 var Nsync_Add_Site = {
 	
 	onReady : function() {
 		
-		var cache = {},
-			lastXhr;
-		jQuery( "#nsync-add-site" ).autocomplete({
-			minLength: 2,
-			source: function( request, response ) {
-				var term = request.term;
-				if ( term in cache ) {
-					response( cache[ term ] );
-					return;
-				}
-				
-				lastXhr = jQuery.getJSON( ajaxurl+"?action=nsync_lookup_site", request, function( data, status, xhr ) {
-					
-					cache[ term ] = data;
-					if ( xhr === lastXhr ) {
-						response( data );
-					}
-				});
-			},
-			select: function(event, ui) { 
+		// trigger ajax search call
+		jQuery( "#nsync-add-site" ).click( function() {
+		
+			var search = jQuery( "#nsync-site-search" ).val();
+			var html = "";
+			
+			var data = {
+					'action': 'nsync_lookup_site',
+					'term': search 
+				};
 				
-				var html = '<label><input name="nsync_options[active][]" type="checkbox" checked="checked" value="'+ui.item.value+'" /> '+ui.item.label+'</label><br />';
+			if (search != "") {
 				
-				jQuery('#select-site').append(html);
-				jQuery( "#nsync-add-site" ).val( '' );
-			 }
+				jQuery.ajax({ 
+					     type : "post",
+					     dataType : "json",
+					     url : "admin-ajax.php",
+					     data : data,
+					     success: function(response) {
+					    	if (response != null) {
+							     jQuery.each(response, function( i, site ) {
+							    	 html += '<div class="search-result-item"><label><input name="nsync_options[active][]" type="checkbox" checked="checked" value="' + site.value + '" /> ' + site.label + '</label></div>';
+								     jQuery('#search-site-results').html(html).fadeIn();
+								  });
+							     
+							     jQuery(".check").fadeIn();
+					    	} else {
+					    		 jQuery(".check").fadeOut();
+								 jQuery('#search-site-results').html("<p> No Results </p>").fadeIn();
+					    	}
+						 },
+						 error: function(){
+	
+							 jQuery(".check").fadeOut();
+							 jQuery('#search-site-results').html("<p> Error Occured </p>").fadeIn();
```

Comment:
How about maybe including what error it is in the message here? The error method receives 3 args; (jqXHR) `jqXHR`, (String) `textStatus`, and (String) `errorThrown` - you could perhaps output the error so we know exactly what the issue was.